app_dart/lib/src/service/branch_service.dart

```diff
@@ -119,15 +119,15 @@ class BranchService {
   /// Latest beta and stable branches are retrieved based on 'beta' and 'stable' tags. Dev branch is retrived
   /// as the latest flutter candidate branch.
   Future<List<Map<String, String>>> getReleaseBranches(
-      {required gh.GitHub github, required gh.RepositorySlug slug}) async {
-    final String betaSha = (await github.repositories.getBranch(slug, 'beta')).commit!.sha!;
-    final String stableSha = (await github.repositories.getBranch(slug, 'stable')).commit!.sha!;
-
-    List<gh.Branch> branches = await github.repositories.listBranches(slug).toList();
-    final String devName = await _getDevBranch(github: github, slug: slug, branches: branches);
-    final String betaName = branches.where(((gh.Branch b) => b.commit!.sha == betaSha)).single.name!;
-    final String stableName = branches.where(((gh.Branch b) => b.commit!.sha == stableSha)).single.name!;
-
+      {required GithubService githubService, required gh.RepositorySlug slug}) async {
+    List<gh.Branch> branches = await githubService.github.repositories.listBranches(slug).toList();
+    final String devName = await _getDevBranch(github: githubService.github, slug: slug, branches: branches);
+
+    final String betaName =
+        (await githubService.getFileContent(slug, 'bin/internal/release-candidate-branch.version', ref: 'beta')).trim();
```

Comment:
Since beta and stable both perform this logic, we should move it into a helper function

Reply:
Sure, refactored. Yeah previously i didn't create a helper function cause I thought they are one liners.